src/main/java/com/woowacamp/soolsool/core/cart/domain/Cart.java

```diff
@@ -1,11 +1,68 @@
 package com.woowacamp.soolsool.core.cart.domain;
 
-import java.util.Set;
-import lombok.RequiredArgsConstructor;
+import com.woowacamp.soolsool.core.cart.code.CartItemErrorCode;
+import com.woowacamp.soolsool.global.exception.SoolSoolException;
+import java.util.List;
+import java.util.Objects;
+import lombok.Getter;
 
-@RequiredArgsConstructor // TODO: 컴파일 에러 방지용 임시 생성자
+@Getter
 public class Cart {
 
+    private static final int MAX_CART_SIZE = 100;
+
     private final Long memberId;
-    private final Set<CartItem> cartItems;
+    private final List<CartItem> cartItems;
+
+    public Cart(final Long memberId, final List<CartItem> cartItems) {
+        validateMember(memberId, cartItems);
+
+        this.memberId = memberId;
+        this.cartItems = cartItems;
+    }
+
+    private void validateMember(final Long memberId, final List<CartItem> cartItems) {
+        boolean hasNotSameMember = cartItems.stream()
+            .anyMatch(cartItem -> !Objects.equals(memberId, cartItem.getMemberId()));
+
+        if (hasNotSameMember) {
+            throw new SoolSoolException(CartItemErrorCode.NOT_EQUALS_MEMBER);
+        }
+    }
+
+    public void addCartItem(final CartItem newCartItem) {
+        validateNull(newCartItem);
+        validateExceedMaxSize();
+        validateDuplicated(newCartItem);
+        validateLiquorStatus(newCartItem);
+
+        cartItems.add(newCartItem);
+    }
+
+    private void validateNull(final CartItem newCartItem) {
+        if (newCartItem == null) {
+            throw new SoolSoolException(CartItemErrorCode.NULL_LIQUOR);
+        }
+    }
```

Comment:
```suggestion
    private void validateNull(final CartItem newCartItem) {
        if (Objects.isNull(newCartItem)) {
            throw new SoolSoolException(CartItemErrorCode.NULL_LIQUOR);
        }
    }
```

키보드에서 null을 입력하면 죄책감이 드는데 이건 어떤가요?

Reply:
오오 좋은 습관인 것 같습니다. 바로 학습 들어갑니다.